cli/internal/cache/cache_http.go

```diff
@@ -152,9 +175,13 @@ func (cache *httpCache) exists(hash string) (bool, int, error) {
 	return true, duration, err
 }
 
-func (cache *httpCache) retrieve(hash string) (bool, []turbopath.AnchoredSystemPath, int, error) {
+func (cache *httpCache) retrieve(
+	hash string,
+	prefixedUI *cli.PrefixedUi,
+) (bool, []turbopath.AnchoredSystemPath, int, error) {
 	resp, err := cache.client.FetchArtifact(hash)
 	if err != nil {
+		prefixedUI.Output(err.Error())
```

Comment:
Is this the only material change? 

You'll want to do something like this:

```suggestion
		prefixedUI.Error(fmt.Sprintf("error fetching from remote cache: %s", err))
```